src/tools/tools/Logger.fs

```diff
@@ -363,56 +367,23 @@ type Logger internal ()=
                                                           logger 
                                                           with get, set
 
-    /// Global default logging level
-    static member val DefaultLogLevel : LogLevel = LogLevel.Info with get, set
-    
-    static member private ParseLogLevel (s:string) =
-        match s with
-            | Prefixi "0" _ 
-            | Prefixi "fatal" _ -> LogLevel.Fatal
-            | Prefixi "1" _
-            | Prefixi "err" _ -> LogLevel.Error
-            | Prefixi "2" _
-            | Prefixi "warn" _ -> LogLevel.Warning
-            | Prefixi "3" _
-            | Prefixi "info" _ -> LogLevel.Info
-            | Prefixi "4" _
-            | Prefixi "mild" _ -> LogLevel.MildVerbose
-            | Prefixi "5" _
-            | Prefixi "med" _  -> LogLevel.MediumVerbose
-            | Prefixi "6" _
-            | Prefixi "wild" _ -> LogLevel.WildVerbose
-            | Prefixi "7" _
-            | Prefixi "extreme" _ -> LogLevel.ExtremeVerbose
-            | _ ->   failwith (sprintf "Unsupported log level: %s" s)
-
-    static member val CommonUsage = "-verbose logLevel  : set default log level.\n"
-
     /// Parse the arguments that configure the behavior of the logger
     static member ParseArgs(args : string[]) =
-        // Parse the generic arguments
-        let mutable i = 0
-        while i < args.Length do
-            match args.[i] with 
-            | Prefixi "-verbose" _ 
-                -> args.[i] <- String.Empty
-                   if i+1 < args.Length then 
-                      Logger.DefaultLogLevel <- Logger.ParseLogLevel (args.[i+1])
-                      args.[i + 1] <- String.Empty
-                   else
-                      failwith "Incorrect Arguments"
-            | _ -> ()
-            i <- i + 1
-        // Parse LoggerProvider's specific arguments
         Logger.LoggerProvider.ParseArgs(args)
 
     static member PrintArgsUsage() =
-        let result = Logger.CommonUsage +  Logger.LoggerProvider.GetArgsUsage()
+        let result = Logger.LoggerProvider.GetArgsUsage()
         printfn "%s" result
 
+    static member DefaultLogLevel() =
+        Logger.LoggerProvider.MaxLogLevelEnabled(Logger.DefaultLogId)
+
+    static member DefaultLogLevel(logId) =
+        Logger.LoggerProvider.MaxLogLevelEnabled(logId)
+
     /// Log "message" if logLevel <= Logger.DefaultLogLevel                                                                    
     static member inline Log(logLevel : LogLevel, message : string) =
-        if logLevel <= Logger.DefaultLogLevel then
+        if Logger.LoggerProvider.IsEnabled(Logger.DefaultLogId, logLevel) then
```

Comment:
My understanding is that with this change, Log message that are suppressed in the past (e.g., WildVerbose) will go through one more function call of IsEnabled(). Will that cause performance issue? Can you check with Sanjeev on the performance with related to sort before and after the change?

Thanks,